### PR TITLE
Fix train data script

### LINKED ISSUE #153: WizardLM Data Gone (prep data script error)
MSFT removed it, so `prepare_train_data.sh` now errors at the end. Should probably document / phase this out.

Simple trace:
```
Traceback (most recent call last):
  File "/home/nathanl/open-instruct/open_instruct/reformat_datasets.py", line 789, in <module>
    globals()[f"convert_{dataset}_data"](os.path.join(args.raw_data_dir, dataset), os.path.join(args.output_dir, dataset))
  File "/home/nathanl/open-instruct/open_instruct/reformat_datasets.py", line 496, in convert_wizardlm_data
    with open(os.path.join(data_dir, "WizardLM_evol_instruct_V2_143k.json"), "r") as fin:
FileNotFoundError: [Errno 2] No such file or directory: 'data/raw_train/wizardlm/WizardLM_evol_instruct_V2_143k.json'
```

### PR DESCRIPTION
**Pull request by natolambert**
Closes #153, makes it so token isnt needed (can use huggingface-cli), I tested this to retrain OLMo 1.7

### CODE REVIEW

File: scripts/prepare_train_data.sh [hunk 1]
```diff
@@ -77,13 +79,16 @@ python scripts/split_sharegpt_conversations.py \
...
+if huggingface-cli whoami &>/dev/null; then
+    huggingface-cli download GAIR/lima --repo-type dataset --local-dir data/raw_train/lima/
+else; then
+    wget --header="Authorization: Bearer $HF_TOKEN" -P data/raw_train/lima/ https://huggingface.co/datasets/GAIR/lima/raw/main/train.jsonl
+fi
 
 echo "Downloading WizardLM dataset..."
-wget -P data/raw_train/wizardlm/ https://huggingface.co/datasets/WizardLM/WizardLM_evol_instruct_V2_196k/resolve/main/WizardLM_evol_instruct_V2_143k.json
+# original data removed wget -P data/raw_train/wizardlm/ https://huggingface.co/datasets/WizardLM/WizardLM_evol_instruct_V2_196k/resolve/main/WizardLM_evol_instruct_V2_143k.json
+wget -P data/raw_train/wizardlm/ https://huggingface.co/datasets/Leon-Leee/Wizardlm_Evol_Instruct_v2_196K_backuped/resolve/main/data/train-00000-of-00001-004cd1ba9dc05e6c.parquet
```
Comment (hamishivi) [inline scripts/prepare_train_data.sh]: Have you verified this is the same data?

**Comment by hamishivi** [comment]
LGTM so long as you've verified the data is the same as our released mixture.

**Comment by natolambert** [comment]
Code for comparing diffs:

```
from datasets import load_dataset

def load_and_compare_datasets(dataset_name1, dataset_name2, split='train'):
    # Load datasets
    dataset1 = load_dataset(dataset_name1, split=split)
    dataset2 = load_dataset(dataset_name2, split=split)

    # Create dictionaries indexed by 'id'
    dict1 = {row['id']: row for row in dataset1}
    dict2 = {row['id']: row for row in dataset2}

    # Find unique ids in each dataset
    unique_ids_to_set1 = set(dict1.keys()) - set(dict2.keys())
    unique_ids_to_set2 = set(dict2.keys()) - set(dict1.keys())

    # Print unique entries
    print("Entries unique to dataset 1:")
    for id in unique_ids_to_set1:
        print(dict1[id])

    print("\nEntries unique to dataset 2:")
    for id in unique_ids_to_set2:
        print(dict2[id])

# Example usage with dataset names and optional split specification
load_and_compare_datasets('ai2-adapt-dev/tulu2-tmp', 'allenai/tulu-v2-sft-mixture', split='train')
```

**Comment by natolambert** [comment]
There are minor differences, mostly via wizardlm taking data down and maybe slight changes to the dataset processing script from the [v2 version on huggingface](https://huggingface.co/datasets/allenai/tulu-v2-sft-mixture). New backups made:
* Tulu 2 as of now: https://huggingface.co/datasets/ai2-adapt-dev/tulu2-tmp
* WizardLM from original tulu 2 mix: https://huggingface.co/datasets/ai2-adapt-dev/wizardlm-backup

**Comment by hamishivi** [comment]
So now the script points to this backup wizardlm version that yields some small differences when you run the train prep script?

**Comment by natolambert** [comment]
@hamishivi, yes.
Backup was taken from your nfs raw train files. I can get the date it was last modified, but I think there are multiple things that have potentially changed since we uploaded tulu v2 sft mix

**Comment by cbfcbf** [comment]
I found a typo at line 86 reformat_dataset.py :     "if num_few_shot_examples > 0" should be "if num_zero_shot_examples > 0"

**Comment by hamishivi** [comment]
Okay, I think this is okay to merge. I ran the script through without errors. I'll add a note saying a samples have shifted, pointing to this PR. Since we have the mixture we actually trained on uploaded already, I think this is okay.
Also fixed up the bug, thanks!

**Comment by natolambert** [comment]
Needs the special approval button :)

**Review by hamishivi — APPROVED** [review]

**Pull request merged**